src/lib.rs

```diff
@@ -481,43 +481,45 @@ fn derive_where_internal(mut item: DeriveInput) -> Result<TokenStream, syn::Erro
 
 	// Search for `crate` option.
 	for attr in &item.attrs {
-		if attr.path.is_ident(DERIVE_WHERE) {
-			if let Ok(Meta::List(list)) = attr.parse_meta() {
-				if list.nested.len() == 1 {
-					if let NestedMeta::Meta(meta) = list
-						.nested
-						.into_iter()
-						.next()
-						.expect("unexpected empty list")
-					{
+		if attr.path().is_ident(DERIVE_WHERE) {
+			if let Meta::List(list) = &attr.meta {
+				if let Ok(nested) =
+					list.parse_args_with(Punctuated::<Meta, Token![,]>::parse_terminated)
+				{
+					if nested.len() == 1 {
+						let meta = nested.into_iter().next().expect("unexpected empty list");
+
 						if meta.path().is_ident("crate") {
 							if let Meta::NameValue(name_value) = meta {
-								if let Lit::Str(lit_str) = &name_value.lit {
-									match lit_str.parse::<Path>() {
-										Ok(path) => {
-											if path == util::path_from_strs(&[DERIVE_WHERE]) {
-												return Err(Error::path_unnecessary(
-													path.span(),
-													&format!("::{}", DERIVE_WHERE),
-												));
-											}
-
-											match crate_ {
-												Some(_) => {
-													return Err(Error::option_duplicate(
-														name_value.span(),
-														"crate",
-													))
-												}
-												None => crate_ = Some(path),
-											}
-										}
+								let path = match &name_value.value {
+									Expr::Lit(ExprLit {
+										lit: Lit::Str(lit_str),
+										..
+									}) => match lit_str.parse::<Path>() {
+										Ok(path) => path,
```

Comment:
We could do a warning, that it is deprecated on nightly IIRC, but probably not worth it

Reply:
Last time I checked it's still not possible to insert warnings inside a proc-macro, so it would have to be an error.
I'm against an error, but a warning would be nice even on non-nightly.

Reply:
Well there is on nightly: https://doc.rust-lang.org/stable/proc_macro/struct.Diagnostic.html#method.span_warning, but not sure how well that works.

Reply:
Ah, right. Never tried it too. In any case, I'm fine with not adding a nightly-only warning.